### ISSUE TITLE
Crash when hovering over Quantum Catcher with bee in it. 

### ISSUE THREAD

**Issue by Epic428** [stal111/Forbidden-Arcanus#106]
When using the Quantum Catcher on a Vanilla Bee or any bee that extends `BeeEntity.java` and then hovering over the catcher in the inventory the game crashes.

The crash states it is due to casting a `Client World` to a `Server World`. This is because of the line [here:](https://github.com/stal111/Forbidden-Arcanus/blob/2d20db191881570cc0e30320762e5507906a460a/src/main/java/com/stal111/forbidden_arcanus/item/QuantumCatcherItem.java#L107). What is happening is that the NBT being loaded for the bee in the `readAdditional` method is calling: `this.angerFromTag((ServerWorld)this.world, p_70037_1_);`. It seems you are loading the entity on the client side and the read is trying to cast it server side to get what it needs.

I don't have a fix for you aside from temporarily disabling the ability to catch bees with the Quantum Catcher.

logs included for reference using only this mod and JEI
[latest.log](https://github.com/stal111/Forbidden-Arcanus/files/5370239/latest.log)
[debug.log](https://github.com/stal111/Forbidden-Arcanus/files/5370240/debug.log)

**Comment by andrewsf** [comment]
Same thing happens with **Polar Bear**.

# Steps to reproduce

- New world
- Give yourself a Quantum Catcher and a Polar Bear Spawn Egg
- Spawn a polar bear
- Capture polar bear
- Mouse over the item in an inventory

# Log

Here is the relevant portion of the crash log:

```
Description: Loading entity NBT

java.lang.ClassCastException: net.minecraft.client.world.ClientWorld cannot be cast to net.minecraft.world.server.ServerWorld
	at net.minecraft.entity.passive.PolarBearEntity.func_70037_a(SourceFile:123) ~[?:?] {re:classloading}
	at net.minecraft.entity.Entity.func_70020_e(Entity.java:1495) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,xf:fml:shetiphiancore:Entity.getOnPosition,xf:fml:immersiveengineering:IE wire damage,xf:fml:shetiphiancore:Entity.createRunningParticles,xf:fml:randompatches:RandomPatches Entity Transformer,pl:mixin:APP:appliedenergistics2.mixins.json:spatial.EntityMixin,pl:mixin:A}
	at net.minecraft.entity.EntityType.lambda$loadEntityUnchecked$2(EntityType.java:458) ~[?:?] {re:classloading,re:mixin}
	at net.minecraft.entity.EntityType$$Lambda$21998/1944702627.accept(Unknown Source) ~[?:?] {}
	at net.minecraft.util.Util.func_215077_a(SourceFile:402) ~[?:?] {re:classloading}
	at net.minecraft.entity.EntityType.func_220330_a(EntityType.java:455) ~[?:?] {re:classloading,re:mixin}
	at com.stal111.forbidden_arcanus.item.QuantumCatcherItem.getEntity(QuantumCatcherItem.java:107) ~[?:1.16.3-1.0.1] {re:classloading}
	at com.stal111.forbidden_arcanus.item.QuantumCatcherItem.func_77624_a(QuantumCatcherItem.java:70) ~[?:1.16.3-1.0.1] {re:classloading}
	at net.minecraft.item.ItemStack.func_82840_a(ItemStack.java:575) ~[?:?] {re:mixin,re:classloading,xf:fml:apotheosis:apothitemusehook,pl:mixin:APP:dankstorage.mixins.json:ItemStackMixin,pl:mixin:APP:forbidden_arcanus.mixins.json:ItemStackMixin,pl:mixin:A}
	at net.minecraft.client.gui.screen.Screen.func_231151_a_(Screen.java:123) ~[?:?] {re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screen.Screen.func_230457_a_(Screen.java:118) ~[?:?] {re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screen.inventory.ContainerScreen.func_230459_a_(ContainerScreen.java:164) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screen.inventory.ChestScreen.func_230430_a_(SourceFile:31) ~[?:?] {re:classloading}
	at net.minecraftforge.client.ForgeHooksClient.drawScreen(ForgeHooksClient.java:281) ~[?:?] {re:classloading}
	at net.minecraft.client.renderer.GameRenderer.func_195458_a(GameRenderer.java:491) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:953) [?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:mixin:APP:kubejs.mixins.json:MinecraftMixin,pl:mixin:APP:shutupexperimentalsettings.mixins.json:client.MixinMinecraft,pl:mixin:APP:assets/botania/botania.mixins.json:AccessorMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:584) [?:?] {re:mixin,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,xf:fml:randompatches:RandomPatches Minecraft Transformer,pl:mixin:APP:kubejs.mixins.json:MinecraftMixin,pl:mixin:APP:shutupexperimentalsettings.mixins.json:client.MixinMinecraft,pl:mixin:APP:assets/botania/botania.mixins.json:AccessorMinecraft,pl:mixin:A,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:184) [?:?] {re:classloading,pl:runtimedistcleaner:A}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_51] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_51] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_51] {}
	at java.lang.reflect.Method.invoke(Method.java:497) ~[?:1.8.0_51] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51) [forge-1.16.3-34.1.24.jar:34.1] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider$$Lambda$476/1124131392.call(Unknown Source) [forge-1.16.3-34.1.24.jar:34.1] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-7.0.1.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-7.0.1.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-7.0.1.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:81) [modlauncher-7.0.1.jar:?] {re:classloading}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:65) [modlauncher-7.0.1.jar:?] {re:classloading}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.entity.passive.PolarBearEntity.func_70037_a(SourceFile:123) ~[?:?] {re:classloading}
-- Entity being loaded --
Details:
	Entity Type: minecraft:polar_bear (net.minecraft.entity.passive.PolarBearEntity)
	Entity ID: 40588
	Entity Name: Polar Bear
	Entity's Exact location: 893.10, 63.00, -1004.35
	Entity's Block location: World: (893,63,-1005), Chunk: (at 13,3,3 in 55,-63; contains blocks 880,0,-1008 to 895,255,-993), Region: (1,-2; contains chunks 32,-64 to 63,-33, blocks 512,0,-1024 to 1023,255,-513)
	Entity's Momentum: 0.00, -0.08, 0.00
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null
Stacktrace:
	at net.minecraft.entity.Entity.func_70020_e(Entity.java:1495) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,xf:fml:shetiphiancore:Entity.getOnPosition,xf:fml:immersiveengineering:IE wire damage,xf:fml:shetiphiancore:Entity.createRunningParticles,xf:fml:randompatches:RandomPatches Entity Transformer,pl:mixin:APP:appliedenergistics2.mixins.json:spatial.EntityMixin,pl:mixin:A}
	at net.minecraft.entity.EntityType.lambda$loadEntityUnchecked$2(EntityType.java:458) ~[?:?] {re:classloading,re:mixin}
	at net.minecraft.entity.EntityType$$Lambda$21998/1944702627.accept(Unknown Source) ~[?:?] {}
	at net.minecraft.util.Util.func_215077_a(SourceFile:402) ~[?:?] {re:classloading}
	at net.minecraft.entity.EntityType.func_220330_a(EntityType.java:455) ~[?:?] {re:classloading,re:mixin}
	at com.stal111.forbidden_arcanus.item.QuantumCatcherItem.getEntity(QuantumCatcherItem.java:107) ~[?:1.16.3-1.0.1] {re:classloading}
	at com.stal111.forbidden_arcanus.item.QuantumCatcherItem.func_77624_a(QuantumCatcherItem.java:70) ~[?:1.16.3-1.0.1] {re:classloading}
	at net.minecraft.item.ItemStack.func_82840_a(ItemStack.java:575) ~[?:?] {re:mixin,re:classloading,xf:fml:apotheosis:apothitemusehook,pl:mixin:APP:dankstorage.mixins.json:ItemStackMixin,pl:mixin:APP:forbidden_arcanus.mixins.json:ItemStackMixin,pl:mixin:A}
	at net.minecraft.client.gui.screen.Screen.func_231151_a_(Screen.java:123) ~[?:?] {re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screen.Screen.func_230457_a_(Screen.java:118) ~[?:?] {re:computing_frames,pl:accesstransformer:B,pl:runtimedistcleaner:A,re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screen.inventory.ContainerScreen.func_230459_a_(ContainerScreen.java:164) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.gui.screen.inventory.ChestScreen.func_230430_a_(SourceFile:31) ~[?:?] {re:classloading}
	at net.minecraftforge.client.ForgeHooksClient.drawScreen(ForgeHooksClient.java:281) ~[?:?] {re:classloading}
```

Encountered the crash w/ AllTheMods 1.1.6b. I did not test with just this mod.

Mod JAR is forbidden_arcanus-1.16.3-1.0.1.jar.

**Comment by Epic428** [comment]
It's a vanilla issue that's causing it. One probe also has the issue: https://github.com/McJtyMods/TheOneProbe/issues/419 (link to vanilla bug report in TOP post)

**Comment by stal111** [comment]
Fixed in 1.16.3-1.0.2
